### PR TITLE
[tools/onert_train] Introduce DataLoader class

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces DataLoader class and inherit it from RawDataLoader.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft: #12265

### CODE REVIEW

**Review by hseok-oh — APPROVED** [review]
LGTM

File: tests/tools/onert_train/src/rawdataloader.cc [hunk 1]
```diff
@@ -39,64 +40,76 @@ uint64_t getRawTensorSize(const std::vector<nnfw_tensorinfo> &infos)
...
-  uint32_t data_length = input_data_length;
+  _data_length = input_data_length;
+}
+
+std::tuple<Generator, uint32_t> RawDataLoader::loadData(const uint32_t batch_size, const float from,
+                                                        const float to)
+{
+  assert(from >= 0.f && from <= 1.f);
+  assert(to >= 0.f && to <= 1.f);
+  assert(from <= to);
```
Comment (glistening) [inline tests/tools/onert_train/src/rawdataloader.cc:73]:
```suggestion
  assert(from >= 0.f);
  assert(from <= to);
  assert(to <= 1.f);
```
(optional) Some conditions can be removed.

**Comment by hseok-oh** [comment]
@nnfw-bot test all

**Review by SlavikMIPT — APPROVED** [review]
LGTM

File: tests/tools/onert_train/src/rawdataloader.cc (hunk 1 above)
Comment (jyoungyun) [reply tests/tools/onert_train/src/rawdataloader.cc:73]: Thank you for suggestion. I will apply it to another pr.

**Pull request merged**